pkg/kubectl/cmd/delete.go

```diff
@@ -95,8 +96,44 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return err
 	}
 
+	// By default use a reaper to delete all related resources.
+	if cmdutil.GetFlagBool(cmd, "cascade") {
+		err := ReapResult(r, f, out)
+		// If the error is something other than "not found" or the user explicitly requested "cascade" error out.
+		if !kubectl.IsNoSuchReaperError(err) || cmd.Flags().Lookup("cascade").Changed {
```

Comment:
Commands can act on multiple objects of multiple different kinds, and visitors may return multiple errors.

Shouldn't we ignore NoSuchReaper errors in ReapResult rather than here?